### PR TITLE
LLVM 17 port

### PR DESCRIPTION
**Pull request by Joonari**

### CODE REVIEW

File: openasip/src/applibs/LLVMBackend/LLVMTCERISCVIntrinsicsLowering.cc [hunk 1]
```diff
@@ -148,7 +151,13 @@ LLVMTCERISCVIntrinsicsLowering::findRegIndexes(
     for (unsigned o = 0; o < it->getNumOperands(); o++) {
         const MachineOperand& mo = it->getOperand(o);
         if (mo.isReg()) {
-            int idx = mo.getReg() - 40;
+            // TODO: Fix this virtual to physical reg mapping
+            #ifdef LLVM_OLDER_THAN_17
```
Comment (pjaaskel) [inline openasip/src/applibs/LLVMBackend/LLVMTCERISCVIntrinsicsLowering.cc:155]: Cpp macro # should be put to the beginning of the line.

File: openasip/src/applibs/LLVMBackend/TCETargetMachine.cc [hunk 2]
```diff
@@ -294,7 +299,7 @@ TCEPassConfig::addPreISel() {
     if (OptLevel != CodeGenOpt::None) {
         // get some pass lists from llvm/Support/StandardPasses.h from 
         // createStandardLTOPasses function. (do not add memcpyopt or dce!)
-        addPass(createInternalizePass());
+        //addPass(createInternalizePass());
```
Comment (pjaaskel) [inline openasip/src/applibs/LLVMBackend/TCETargetMachine.cc:302]: If it's not needed, just remove it compeltely.

File: openasip/src/applibs/LLVMBackend/passes/LowerIntrinsics.cc [hunk 3]
```diff
@@ -46,7 +47,10 @@ IGNORE_COMPILER_WARNING("-Wunused-parameter")
 #include "llvm/IR/DerivedTypes.h"
 #include "llvm/IR/LLVMContext.h"
 #include "llvm/Support/Compiler.h"
+#include "llvm/IR/Function.h"
```
Comment (pjaaskel) [inline openasip/src/applibs/LLVMBackend/passes/LowerIntrinsics.cc:50]: These includes should use <>

**Review by pjaaskel — CHANGES_REQUESTED** [review]
Well done. I'll test.

File: openasip/src/applibs/LLVMBackend/TCETargetMachine.cc (hunk 2 above)
Comment (Joonari) [reply openasip/src/applibs/LLVMBackend/TCETargetMachine.cc]: Removed.

File: openasip/src/applibs/LLVMBackend/passes/LowerIntrinsics.cc (hunk 3 above)
Comment (Joonari) [reply openasip/src/applibs/LLVMBackend/passes/LowerIntrinsics.cc]: Fixed

**Review by pjaaskel — APPROVED** [review]

**Pull request merged**